src/librustc_mir/transform/mod.rs

```diff
@@ -21,3 +26,32 @@ pub mod deaggregator;
 pub mod instcombine;
 pub mod copy_prop;
 pub mod inline;
+
+pub(crate) fn provide(providers: &mut Providers) {
+    self::qualify_consts::provide(providers);
+    *providers = Providers {
+        ..*providers
+    };
+}
+
+pub(crate) fn run_suite<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
```

Comment:
If you moved those 3 entry points here, you'd definitely not need to make this public.